README.md
[![codecov.io](https://codecov.io/github/Kitware/candela/coverage.svg?branch=master)](https://codecov.io/github/Kitware/candela?branch=master)

# ![](docs/static/favicon.png) Candela

Candela is an open-source suite of interoperable web visualization
components for [Kitware](http://www.kitware.com)'s [Resonant](http://resonant.kitware.com) platform.
Candela focuses on making scalable, rich visualizations available
with a normalized API for use in real-world data science applications.

Please see our documentation at https://candela.readthedocs.io.


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.23.4"></a>
## [0.23.4](https://github.com/kitware/candela/compare/v0.23.3...v0.23.4) (2018-07-12)




**Note:** Version bump only for package candela

<a name="0.23.3"></a>
## [0.23.3](https://github.com/kitware/candela/compare/v0.23.2...v0.23.3) (2018-06-12)




**Note:** Version bump only for package candela

<a name="0.23.2"></a>
## [0.23.2](https://github.com/kitware/candela/compare/v0.23.1...v0.23.2) (2018-06-06)




**Note:** Version bump only for package candela

<a name="0.23.1"></a>
## [0.23.1](https://github.com/kitware/candela/compare/v0.23.0...v0.23.1) (2018-06-04)




**Note:** Version bump only for package candela

<a name="0.23.0"></a>
# [0.23.0](https://github.com/kitware/candela/compare/v0.22.0...v0.23.0) (2018-05-31)


### Features

* **vega:** add `calculateSize` option ([f2added](https://github.com/kitware/candela/commit/f2added))




<a name="0.22.0"></a>
# [0.22.0](https://github.com/kitware/candela/compare/v0.21.0...v0.22.0) (2018-05-16)


### Features

* **jupyterlab:** Add jupyterlab extension and update pycandela ([03c77e8](https://github.com/kitware/candela/commit/03c77e8))




<a name="0.21.0"></a>
# [0.21.0](https://github.com/kitware/candela/compare/v0.20.1...v0.21.0) (2018-05-14)


### Features

* add an all-inclusive bundle package ([3f86d39](https://github.com/kitware/candela/commit/3f86d39))




<a name="0.20.1"></a>
## [0.20.1](https://github.com/kitware/candela/compare/v0.20.0...v0.20.1) (2018-05-10)




**Note:** Version bump only for package candela

<a name="0.20.0"></a>
# [0.20.0](https://github.com/kitware/candela/compare/v0.19.1...v0.20.0) (2018-05-09)


### Features

* enable monorepo ([ed85692](https://github.com/kitware/candela/commit/ed85692))
* enable monorepo ([dd261f9](https://github.com/kitware/candela/commit/dd261f9))
* enable monorepo ([6d26824](https://github.com/kitware/candela/commit/6d26824))
* enable monorepo ([88a5315](https://github.com/kitware/candela/commit/88a5315))
* enable monorepo ([79a107a](https://github.com/kitware/candela/commit/79a107a))
* enable monorepo ([45d693e](https://github.com/kitware/candela/commit/45d693e))
* enable monorepo ([f9571b6](https://github.com/kitware/candela/commit/f9571b6))
* enable monorepo ([ed78052](https://github.com/kitware/candela/commit/ed78052))
* enable monorepo ([72dd74c](https://github.com/kitware/candela/commit/72dd74c))
* enable monorepo ([673990c](https://github.com/kitware/candela/commit/673990c))
* enable monorepo ([110131c](https://github.com/kitware/candela/commit/110131c))
* enable monorepo ([b4e4e00](https://github.com/kitware/candela/commit/b4e4e00))
* enable monorepo ([f28d4ce](https://github.com/kitware/candela/commit/f28d4ce))
* enable monorepo ([e0c2ffa](https://github.com/kitware/candela/commit/e0c2ffa))
* enable monorepo ([145c113](https://github.com/kitware/candela/commit/145c113))
* enable monorepo ([a538cac](https://github.com/kitware/candela/commit/a538cac))


src/main.js
// When all scripts are loaded, we invoke the application
$(function () {
    $('.dash-container').trackerdash({
        name: 'Beta Ground Truth',
        day: '2014-06-03',
        branch: 'master',
        trend: [{
            success: 3,
            bad: 1,
            fail: 6,
            date: '2014-05-27'
        }, {
            success: 2,
            bad: 3,
            fail: 5,
            date: '2014-05-28'
        }, {
            success: 4,
            bad: 5,
            fail: 1,
            date: '2014-05-29'
        }, {
            success: 3,
            bad: 3,
            fail: 4,
            date: '2014-05-30'
        }, {
            success: 7,
            bad: 2,
            fail: 1,
            date: '2014-05-31'
        }, {
            success: 6,
            bad: 4,
            fail: 0,
            date: '2014-06-01'
        }, {
            success: 10,
            bad: 0,
            fail: 0,
            date: '2014-06-02'
        }, {
            success: 8,
            bad: 1,
            fail: 1,
            date: '2014-06-03'
        }],
        totalDatasets: 15,
        totalDistance: 1243,
        percentErrorByDataset: [{
            dataset: 'dataset0.jpg',
            algorithm: 'alga',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset0.jpg',
            algorithm: 'sm',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset0.jpg',
            algorithm: 'ba',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset1',
            algorithm: 'alga',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset1',
            algorithm: 'sm',
            length: 300,
            mean: 15,
            current: 14,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset1',
            algorithm: 'ba',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset2',
            algorithm: 'alga',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset2',
            algorithm: 'sm',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset2',
            algorithm: 'ba',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset3',
            algorithm: 'alga',
            length: 300,
            mean: 11,
            current: 28,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset3',
            algorithm: 'sm',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 50
        }, {
            dataset: 'dataset3',
            algorithm: 'ba',
            length: 300,
            mean: 11,
            current: 5,
            target: 7,
            warning: 10,
            fail: 25,
            max: 27,
            callback: function () { alert('dataset3-ba'); }
        }, {
            dataset: 'dataset2',
            algorithm: 'com',
            length: 300,
            mean: 11,
            current: 3,
            target: 7,
            warning: 10,
            fail: 25,
            max: 27,
            callback: function () { alert('dataset2-com'); }
        }],
        datasetLabelMap: {
            'dataset2': 'rotation',
            'dataset1': 'label'
        },
        datasetMap: {
            'dataset3': 'http://www.trackerdash.com',
            'dataset2': function () { alert('dataset 2'); }
        },
        trajectoryMap: {
            'dataset0.jpg': function () { alert('dataset 0 -- trajectory'); },
            'dataset1': 'https://github.com'
        },
        algorithms: [{
            name: 'Visual-Inertial Odometry',
            abbreviation: 'ALGA'
        }, {
            name: 'ALGB (Pose Graph)',
            abbreviation: 'SM'
        }, {
            name: 'Bundle Adjustment',
            abbreviation: 'BA'
        }, {
            name: 'Concurrent Odometry and Mapping',
            abbreviation: 'ALGC'
        }]
    });
});
